### PR TITLE
[exporter/prometheus] Fix exporter error log missing

### PR DESCRIPTION
**Pull request by ZhouZhongzhi**
Prometheus fails to generate logs when prometheus exporter produced a check exception occurs.

Zap.Logger is wrapped by wrapper mode to implement promhttp.Logger.

### CODE REVIEW

**Comment by ZhouZhongzhi** [comment]
The relevant MergeRequest
#7252

**Comment by ZhouZhongzhi** [comment]
@Aneurysm9

**Comment by ZhouZhongzhi** [comment]
Conflicts with the Node_exporter version cannot be found without a log
You're just going to see some indicators missing
But don't know why

@Aneurysm9 

https://github.com/prometheus/client_golang/issues/984

File: CHANGELOG.md [hunk 1]
```diff
@@ -312,7 +312,7 @@
   function are now unexported. (#TBD)
 - `newrelicexporter` marked as deprecated (#7284)
 
-### 🚀 New components 🚀
+- `mdatagen`: Fix validation of `enabled` field in metadata.yaml (#7166)
```
Comment (pmm-sumo) [inline CHANGELOG.md:315]: Why this change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -323,6 +323,7 @@
 - `k8sattributeprocessor`: Parse IP out of net.Addr to correctly tag k8s.pod.ip (#7077)
 - `k8sattributeprocessor`: Process IP correctly for net.Addr instances that are not typed (#7133)
 - `mdatagen`: Fix validation of `enabled` field in metadata.yaml (#7166)
+- `prometheusexporter`: Prometheus fails to generate logs when prometheus exporter produced a check exception occurs. (#8949)
```
Comment (pmm-sumo) [inline CHANGELOG.md:326]: This should be at the top of the changelog (under the upcoming release)

File: CHANGELOG.md (hunk 1 above)
Comment (ZhouZhongzhi) [reply CHANGELOG.md]: Code merge error，modified

File: CHANGELOG.md (hunk 2 above)
Comment (ZhouZhongzhi) [reply CHANGELOG.md]: Has been modified

File: CHANGELOG.md [hunk 3]
```diff
@@ -320,6 +320,7 @@
 
 ### 🧰 Bug fixes 🧰
 
+- `prometheusexporter`: Prometheus fails to generate logs when prometheus exporter produced a check exception occurs. (#8949)
```
[the remark this answers is not in the document]
Comment (ZhouZhongzhi) [reply CHANGELOG.md]: Has been modified

[the remark this answers is not in the document]
Comment (pmm-sumo) [reply CHANGELOG.md]: Great, thank you

**Comment by ZhouZhongzhi** [comment]
@Aneurysm9

**Comment by pmm-sumo** [comment]
Could you resolve conflicts @ZhouZhongzhi ? I think it looks good otherwise

File: CHANGELOG.md [hunk 4]
```diff
@@ -87,6 +87,7 @@
 ### 🚀 New components 🚀
 
 - `sigv4authextension`: Enable component (#8518)
+- `prometheusexporter`: Prometheus fails to generate logs when prometheus exporter produced a check exception occurs. (#8949)
```
Comment (pmm-sumo) [inline CHANGELOG.md:90]: This should be under "## Unreleased", "Bug Fixes". So around line 43

Comment (ZhouZhongzhi) [reply CHANGELOG.md]: I have completed the merger

**Comment by ZhouZhongzhi** [comment]
@Aneurysm9 
Can you help me merge this feature into main branch?
This mergeRequest has been going on for a few months now

**Comment by ZhouZhongzhi** [comment]
@Aneurysm9

File: CHANGELOG.md [hunk 5]
```diff
@@ -73,6 +73,7 @@
 - `signalfxexporter`: Fix bug to enable timeouts for correlating traces and metrics (#9101)
 - `windowsperfcountersreceiver`: fix exported values being integers instead of doubles (#9138)
 - `prometheusreceiver`: Fix issues with relabelling the `job` and `instance` labels. (#8780)
+- `prometheusexporter`: Prometheus fails to generate logs when prometheus exporter produced a check exception occurs. (#8949)
```
Comment (pmm-sumo) [inline CHANGELOG.md:76]: Since we released a new version in the meantime, this should go to line 19 or so

**Comment by pmm-sumo** [comment]
@dashpole I think you were making some updates to prometheus receiver recently. Could you take a look if this PR looks good?

**Review by pmm-sumo — APPROVED** [review]

**Review by dashpole — APPROVED** [review]
good find

**Review by codeboten — APPROVED** [review]
